New option: silent mode

adamkiss opened the pull request:
Hi,

thanks for this great gulp plugin. Unfortunately, it's far too talkative for me :smile:, so I added a 'silent' option. And since I'd like to have at least *some* feedback, I added a 'totals' output.

Changes:
* silence all console messages
* count processed files (and basic statistics: matched/unmatched, etc.)
* output a result of processing: 'Created X images (matched Y of Z images)'
* If 0 files were matched, be completely silent.

All tests pass (well, except --webp, because I don't have libvips with webp compiled), and the silent mode must be activated (original verbose mode is the default mode) so it's completely backwards compatible.

mahnunchik commented:
Hi @adamkiss 

Thanks for your PR.

I have some points to be discussed:
* I totally agree with `silent` option. It should be really helpful in some cases.
* But I think it would be better to have separate option (for example `statistics`) for printing a result.
* As well as output a result when 0 files has been processed.

adamkiss commented:
Hm, I'm okay with that. I'd like to have current silent+warn option, but didn't want to introduce too many changes.

How would you like to proceed? `verbosity` string with multiple values (`silent`, `warn`, `all`), or multiple boolean options?

(in the code, I think we'll parse the string or subset of options into booleans anyway.

mahnunchik commented:
In general it is not logging levels so I don't see any reason to introduce `verbosity` string. 

I prefer to have two boolean options:
* Show or not a process of manipulations (`silent`, `false` by default)
* Show or not a result of manipulations (I think it should be `true` by default)

adamkiss commented:
Well, it *is* logging, except not to file, but stdout.

I'll look into the upscale warns to see if I think it's a meaningful option, otherwise I'll just add a `showStats` option, which will be `true` by default.

adamkiss commented:
Right, so I added `options.stats` and `options.silent` with default values as we agreed, and added a notes about them to readme.

All tests pass, so check it out, and if it's okay, bump & release.

👍🏻

(edit: I mistakingly comitted my preferred settings. Fixed)

mahnunchik commented:
Hi @adamkiss 

Thanks you!
Landed in version 2.2.0.

I really like the statistics :wink: 
```bash
[22:45:20] gulp-responsive: Created 73 images (matched 10 of 23 images)
```
